ceremonies/src/lib.rs

```diff
@@ -672,13 +678,19 @@ impl<T: Config> Module<T> {
 		assignment_params: AssignmentParams,
 		n: u64,
 	) -> Result<MeetupIndexType, Error<T>> {
-		let index = (participant_index
-			.checked_mul(assignment_params.s1)
-			.ok_or(Error::<T>::CheckedMath)?
-			.checked_add(assignment_params.s2)
-			.ok_or(Error::<T>::CheckedMath)? %
-			assignment_params.m) %
-			n;
+		let index = checked_modulo(
+			checked_modulo(
+				participant_index
+					.checked_mul(assignment_params.s1)
+					.ok_or(Error::<T>::CheckedMath)?
+					.checked_add(assignment_params.s2)
+					.ok_or(Error::<T>::CheckedMath)?,
+				assignment_params.m,
+			)
+			.ok_or(Error::<T>::CheckedMath)?,
+			n,
+		)
+		.ok_or(Error::<T>::CheckedMath)?;
 
```

Comment:
I see you copied not so beautiful code from my side. Allow me to introduce some beatification.

```
		// Simple closure that returns an option to handily use the `?` operator.
		let assign = || -> Option<MeetupIndexType> {
			participant_index
				.checked_mul(assignment_params.s1)?
				.checked_add(assignment_params.s2)
				.map(|div| checked_modulo(div, assignment_params.m))?
				.map(|div| checked_modulo(div, n))?
		};

		assign().ok_or(Error::<T>::CheckedMath)
```